sdk/keyvault/azure_security_keyvault_secrets/README.md

```diff
@@ -1,24 +1,429 @@
+
+[Source code] | [Package (crates.io)] | [API reference documentation] | [Product documentation]
 
 ## Getting started
 
+### Install the package
+
+Install the Azure Key Vault Secrets client library for Rust with [Cargo]:
+
+```sh
+cargo add azure_security_keyvault_secrets
+```
+
+### Prerequisites
+
+* An [Azure subscription].
+* An existing Azure Key Vault. If you need to create an Azure Key Vault, you can use the Azure Portal or [Azure CLI].
+* Authorization to an existing Azure Key Vault using either [RBAC] (recommended) or [access control].
+
+If you use the Azure CLI, replace `<your-resource-group-name>` and `<your-key-vault-name>` with your own, unique names:
+
+```azurecli
+az keyvault create --resource-group <your-resource-group-name> --name <your-key-vault-name>
+```
+
+### Install dependencies
+
+Add the following crates to your project:
+
+```sh
+cargo add azure_identity azure_core serde_json tokio
+```
+
+### Authenticate the client
+
+In order to interact with the Azure Key Vault service, you'll need to create an instance of the `SecretClient` struct. You need a **vault url**, which you may see as "DNS Name" in the portal, and credentials to instantiate a client object.
+
+The example shown below use a `DefaultAzureCredential`, which is appropriate for most scenarios including local development and production environments. Additionally, we recommend using a managed identity for authentication in production environments. You can find more information on different ways of authenticating and their corresponding credential types in the [Azure Identity] documentation.
+
+The `DefaultAzureCredential` will automatically pick up on an Azure CLI authentication. Ensure you are logged in with the Azure CLI:
+
+```azurecli
+az login
+```
+
+Instantiate a `DefaultAzureCredential` to pass to the client. The same instance of a token credential can be used with multiple clients if they will be authenticating with the same identity.
+
+### Set and Get a Secret
+
+```rust
+use azure_core::{RequestContent, Response};
+use azure_identity::DefaultAzureCredential;
+use azure_security_keyvault_secrets::{
+    models::{SecretBundle, SecretSetParameters},
+    SecretClient,
+};
+use serde_json::to_vec;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Create a new secret client
+    let credential = DefaultAzureCredential::new()?;
+    let client = SecretClient::new(
+        "https://your-key-vault-name.vault.azure.net/",
+        credential,
+        None,
+    )?;
+
+    // Create a new secret using the secret client.
+    let mut secret_set_parameters = SecretSetParameters::default();
+    secret_set_parameters.value = Some("secret-value".to_string());
```

Comment:
Easier and idiomatic:
```suggestion
    let secret_set_parameters = SecretSetParameters {
        value: Some("secret-value".to_string()),
        ..Default::default()
    };
```

Reply:
Was being blocked by the `#[non_exhaustive]` on this before but I believe it's been removed now.

Reply:
I removed it manually. @jhendrixMSFT were you going to try to do this change before we release beta.1?

Reply:
Yeah it's there now in `v0.8.0` of the emitter.